src/libraries/Common/tests/SingleFileTestRunner/SingleFileTestRunner.cs

```diff
@@ -0,0 +1,72 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using System.Runtime.Loader;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+using Xunit;
+using Xunit.Abstractions;
+using Xunit.Sdk;
+
+public class SingleFileTestRunner : XunitTestFramework
+{
+    private SingleFileTestRunner(IMessageSink messageSink)
+    : base(messageSink) { }
+
+    public static int Main(string[] args)
+    {
+        var asm = typeof(SingleFileTestRunner).Assembly;
+        Console.WriteLine("Running assembly:" + asm.FullName);
+
+        var diagnosticSink = new ConsoleDiagnosticMessageSink();
+        var testsFinished = new TaskCompletionSource();
+        var testSink = new TestMessageSink();
+        var summarySink = new DelegatingExecutionSummarySink(testSink,
+            () => false,
+            (completed, summary) => Console.WriteLine($"Tests run: {summary.Total}, Errors: 0, Failures: {summary.Failed}, Skipped: {summary.Skipped}. Time: {TimeSpan.FromSeconds((double)summary.Time).TotalSeconds}s"));
+        var resultsXmlAssembly = new XElement("assembly");
+        var resultsSink = new DelegatingXmlCreationSink(summarySink, resultsXmlAssembly);
+
+        //testSink.Execution.TestPassedEvent += args => { Console.WriteLine($"[PASS] {args.Message.Test.DisplayName}"); };
+        testSink.Execution.TestSkippedEvent += args => { Console.WriteLine($"[SKIP] {args.Message.Test.DisplayName}"); };
+        testSink.Execution.TestFailedEvent += args => { Console.WriteLine($"[FAIL] {args.Message.Test.DisplayName}{Environment.NewLine}{Xunit.ExceptionUtility.CombineMessages(args.Message)}{Environment.NewLine}{Xunit.ExceptionUtility.CombineStackTraces(args.Message)}"); };
+
+        //testSink.Execution.TestAssemblyStartingEvent += args => { Console.WriteLine($"Running tests for {args.Message.TestAssembly.Assembly}"); };
+        testSink.Execution.TestAssemblyFinishedEvent += args =>
+        {
+            Console.WriteLine($"Finished {args.Message.TestAssembly.Assembly}{Environment.NewLine}");
+            testsFinished.SetResult();
+        };
+
+        var xunitTestFx = new SingleFileTestRunner(diagnosticSink);
+        var asmInfo = Reflector.Wrap(asm);
+        var asmName = asm.GetName();
+        var executor = xunitTestFx.CreateExecutor(asmName);
+
+        executor.RunAll(resultsSink, TestFrameworkOptions.ForDiscovery(), TestFrameworkOptions.ForExecution());
```

Comment:
If we ever end up resurrecting this runner, the magic juju to put instead of these two lines that will work right for tests marked with ActiveIssue or PlatformSpecific is:

```csharp
        var discoverySink = new TestDiscoverySink();
        var discoverer = xunitTestFx.CreateDiscoverer(asmInfo);
        discoverer.Find(false, discoverySink, TestFrameworkOptions.ForDiscovery());
        discoverySink.Finished.WaitOne();

        XunitFilters filters = new XunitFilters();
        filters.ExcludedTraits.Add("category", new List<string> { "failing" });

        var filteredTestCases = discoverySink.TestCases.Where(filters.Filter).ToList();

        var executor = xunitTestFx.CreateExecutor(asmName);
        executor.RunTests(filteredTestCases, resultsSink, TestFrameworkOptions.ForExecution());
```